File: stripe/src/test/java/com/stripe/android/model/ShippingInformationTest.java
package com.stripe.android.model;

import androidx.annotation.NonNull;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ShippingInformationTest {

    @Test
    public void testEquals() {
        assertEquals(createShippingInformation(), createShippingInformation());
    }

    @NonNull
    private ShippingInformation createShippingInformation() {
        return new ShippingInformation(Address.fromString(AddressTest.JSON_ADDRESS),
                "home", "555-123-4567");
    }
}

File: stripe/src/test/java/com/stripe/android/model/ShippingInformationTest.kt
package com.stripe.android.model

import org.junit.Assert.assertEquals
import org.junit.Test

class ShippingInformationTest {

    @Test
    fun testEquals() {
        assertEquals(createShippingInformation(), createShippingInformation())
    }

    @Test
    fun testToParamMapStripsNulls() {
        assertEquals(
            mapOf("name" to "home"),
            ShippingInformation(null, "home", null).toParamMap()
        )
    }

    private fun createShippingInformation(): ShippingInformation {
        return ShippingInformation(Address.fromJson(AddressFixtures.ADDRESS_JSON),
            "home", "555-123-4567")
    }
}
